aws-datastore/src/androidTest/java/com/amplifyframework/datastore/storage/sqlite/SynchronousStorageAdapter.java

```diff
@@ -0,0 +1,279 @@
+
+    private SynchronousStorageAdapter(
+            @NonNull LocalStorageAdapter localStorageAdapter,
+            long operationTimeoutMs) {
+        this.localStorageAdapter = localStorageAdapter;
+        this.operationTimeoutMs = operationTimeoutMs;
+    }
+
+    /**
+     * Creates a new instance which will proxy calls to the provided {@link LocalStorageAdapter}.
+     * The synchronous operations exposed by the returned adapter will timeout after a default,
+     * "reasonable" delay.
+     * @param localStorageAdapter Adapter to which calls will be proxied
+     * @return A SynchronousStorageAdapter configured to proxy towards the provided async storage adapter
+     */
+    public static SynchronousStorageAdapter instance(@NonNull LocalStorageAdapter localStorageAdapter) {
+        Objects.requireNonNull(localStorageAdapter);
+        return new SynchronousStorageAdapter(localStorageAdapter, DEFAULT_OPERATION_TIMEOUT_MS);
+    }
+
+    /**
+     * Creates a new instance which will proxy calls to the provided {@link LocalStorageAdapter}.
+     * The synchronous operations exposed by the returned adapter will timeout after the provided
+     * amount of time, in milliseconds.
+     * @param localStorageAdapter Adapter to which calls will be proxied
+     * @param operationTimeoutMs Amount of time after which an operation will time out
+     * @return A SynchronousStorageAdapter configured to proxy towards the provided async storage adapter
+     */
+    public static SynchronousStorageAdapter instance(
+            @NonNull LocalStorageAdapter localStorageAdapter, long operationTimeoutMs) {
+        Objects.requireNonNull(localStorageAdapter);
+        return new SynchronousStorageAdapter(localStorageAdapter, operationTimeoutMs);
+    }
+
+    /**
+     * Initializes the storage adapter.
+     * @param context An Android Context
+     * @return The list of model schema that are available for use in the adapter
+     * @throws DataStoreException On any initialization failure
+     */
+    @SuppressWarnings("UnusedReturnValue")
+    List<ModelSchema> initialize(@NonNull Context context) throws DataStoreException {
+        return Await.result(
+            operationTimeoutMs,
+            (Consumer<List<ModelSchema>> onResult, Consumer<DataStoreException> onError) ->
+                localStorageAdapter.initialize(context, onResult, onError)
+        );
+    }
+
+    /**
+     * Terminate use of the storage adapter.
+     * @throws DataStoreException On failure to terminate
+     */
+    void terminate() throws DataStoreException {
+        localStorageAdapter.terminate();
+    }
+
+    /**
+     * Save a model into the storage adapter.
+     * @param model Model to save
+     * @param <T> Type of model being saved
+     * @throws DataStoreException On any failure to save model into storage adapter
+     */
+    <T extends Model> void save(@NonNull T model) throws DataStoreException {
+        //noinspection ConstantConditions
```

Comment:
Something I noticed as weird while I was working on the same thing, but why did we do this? `save(T, QueryPredicate)` takes in two non-null values, but we are passing in `null` here and suppressing the warning with `ConstantConditions`. Same thing in other operations.

Reply:
I think ultimately what we need to do is to dis-allow this `null` `QueryPredicate` everywhere, entirely. Right now, when we use a `null` `QueryPredicate`, it means "we don't want to do any filtering; allow everything." So, we could make that explicit by having a special type of non-null query predicate. Maybe called like `AllowAllQueryPredicate` or `NoOpQueryPredicate`. We could make a little utility class, with a factory method: `QueryPredicates.matchAll()`. The thing that this call returns would be one of these special non-null query predicates, that just matches everything.

"Matching everything" would still need to be coded differently for in-memory, SQL, and GraphQL. Right now we have this code, I think it just checked `null == queryPredicate` instead of like `queryPredicate instanceof AllowAllQueryPreciate`, (or, however we might name this new type of predicate.)